File: ts/wam/5134.ts
export const WamMessagingUserJourney = 5134 as const

export interface MessagingUserJourneyEvent {
  /** field 1, wire `app_session_id` */
  appSessionId?: string
  /** field 2, wire `messaging_action_type` */
  messagingActionType?: ACTIONTYPE
  /** field 3, wire `pin_in_chat_expiry_secs` */
  pinInChatExpirySecs?: number
  /** field 4, wire `thread_type` */
  threadType?: THREADTYPE
  /** field 5, wire `ui_surface` */
  uiSurface?: TSSURFACE
  /** field 6, wire `user_journey_funnel_id` */
  userJourneyFunnelId?: string
  /** field 7, wire `user_role` */
  userRole?: USERROLETYPE
  /** field 8, wire `is_self_pin` */
  isSelfPin?: boolean
  /** field 9, wire `media_type` */
  mediaType?: MEDIATYPE
  /** field 10, wire `unified_session_id` */
  unifiedSessionId?: string
}

File: go/wam/5134.go
const WamMessagingUserJourney = 5134 // channel: regular

type MessagingUserJourneyEvent struct {
	AppSessionId string `wam:"app_session_id"` // field 1
	MessagingActionType ACTIONTYPE `wam:"messaging_action_type"` // field 2
	PinInChatExpirySecs int64 `wam:"pin_in_chat_expiry_secs"` // field 3
	ThreadType THREADTYPE `wam:"thread_type"` // field 4
	UiSurface TSSURFACE `wam:"ui_surface"` // field 5
	UserJourneyFunnelId string `wam:"user_journey_funnel_id"` // field 6
	UserRole USERROLETYPE `wam:"user_role"` // field 7
	IsSelfPin bool `wam:"is_self_pin"` // field 8
	MediaType MEDIATYPE `wam:"media_type"` // field 9
	UnifiedSessionId string `wam:"unified_session_id"` // field 10
}